pages/index.vue

```diff
@@ -71,83 +25,7 @@ const isSmallScreen = breakpoints.smallerOrEqual("lg");
     <template v-else>
-      <template v-for="group in discoverGroups" :key="group.header?.id || 0">
-        <PageHeading v-if="group.header" :level="3">
-          <div class="flex items-center justify-between">
-            <div>
-              <a
-                v-if="typeof group.header.link === 'string'"
-                :href="group.header.link"
-              >
-                {{ group.header.title }}
-              </a>
-              <span v-else>{{ group.header.title }}</span>
-            </div>
-            <a v-if="group.header.link" :href="group.header.link">
-              <ButtonStyled intent="secondary" size="small">
-                <span class="whitespace-nowrap">{{
-                  t("home.list.see-all")
-                }}</span></ButtonStyled
-              >
-            </a>
-          </div>
-        </PageHeading>
-        <div
-          v-if="group.header?.useCoverCarousel"
-          class="flex flex-row flex-wrap gap-6"
-        >
-          <template
-            v-for="item in group.items.slice(0, isSmallScreen ? 4 : 6)"
-            :key="item.id"
-          >
-            <NuxtLink
-              v-if="item.type === 'album'"
-              :to="{ name: 'album-id', params: { id: item.id } }"
-            >
-              <ItemCard :item="item" />
-            </NuxtLink>
-            <NuxtLink
-              v-else-if="item.type === 'playlist'"
-              :to="{ name: 'playlist-curated-id', params: { id: item.id } }"
-            >
-              <ItemCard :item="item" />
-            </NuxtLink>
-            <NuxtLink
-              v-else-if="item.type === 'podcast'"
-              :to="{ name: 'playlist-podcast-id', params: { id: item.id } }"
-            >
-              <ItemCard :item="item" />
-            </NuxtLink>
-            <div
-              v-else
-              class="grid w-52 flex-shrink-0 basis-52 gap-4"
-              style="background-color: rgba(255, 0, 0, 0.4); color: red"
-            >
-              "{{ item.type }}" is not yet implemented ...
-            </div>
-          </template>
-        </div>
-        <ol v-else class="w-full divide-y divide-label-separator">
-          <template v-for="item in group.items" :key="item.id">
-            <TrackItem
-              v-if="item.type === 'track'"
-              :track="item"
-              :is-track-type-known="true"
-              show-thumbnail
-              @play-track="playItem(item, group)"
-            ></TrackItem>
-            <ContributorListItem
-              v-else-if="item.type === 'contributor'"
-              :contributor="item"
-            ></ContributorListItem>
-            <li v-else>
-              <div style="background-color: rgba(255, 0, 0, 0.4); color: red">
-                "{{ item.type }}" is not yet implemented ...
-              </div>
-            </li>
-          </template>
-        </ol>
-      </template>
+      <DocumentModels v-if="models !== null" :models="models"></DocumentModels>
```

Comment:
When including the few lines above, this seems to be replacible by `<DocumentList>`. Or was there a reason not to do it, which I overlooked?

Reply:
<DocumentModels ...> is supposed to be used in a lot of cases. E.g. all pages under browse use it to render the documents.
So the page needs to handle loading of documents and DocumentModels renders them.